### PR TITLE
Update docs and add some dev tools

### PR DESCRIPTION
**Pull request by bloodearnest**
- `just token`/`dev-token.py` will generate a token to use
 - add default .env template
 - set default signing key in `__init__.py` so it is always set.

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -28,7 +28,43 @@ just dev_setup
...
+
+
+## Setup config
+
+Config is loaded via env vars from .env:
+
+`cp dotenv-sample .env`
+
+The default dev config will run this service on `http://127.0.0.1:8001`, and
+assume a local job-runner is running on `http://127.0.0.8:8000`
```
Comment (ghickman) [inline DEVELOPERS.md:44]: Oh that's _really_ nice 👌

File: DEVELOPERS.md [hunk 2]
```diff
@@ -28,7 +28,43 @@ just dev_setup
...
+
+
+## Generate token
+
+If you need a test token to access the APIs, you can run:
+
+just token [WORKSPACE]
+
+This will output a token that can be used to access urls for workspace
+WORKSPACE. You do not include workspace, it will be able to access any
```
Comment (ghickman) [inline DEVELOPERS.md:63]: s/You/If you/

File: README.md [hunk 3]
```diff
@@ -1,20 +1,14 @@
 # release-hatch
 
-This is a template for an OpenSAFELY Core repository.
+This provides a small API to authenicate and serve medium-privacy files in TRE
+environments, as well as handle requesting a Release and the reviewing and
+uploading.
 
-Put your project description here.
+It has no database, it just uses the directories and files found on disk.
```
Comment (ghickman) [inline README.md:7]: The latter half of this sentence doesn't sit right with me, how about:

> It has no database, all state is derived from the directories and files found on disk.

File: README.md [hunk 4]
```diff
@@ -1,20 +1,14 @@
...
-- [ ] Does the repo require a Dockerfile?
-  If not, delete:
-  - Dockerfile -
-  - .dockerignore
-  - hadolint pre-commit hook from `.pre-commit-config.yaml`
-  - `lint-dockerfile` action from `.github/workflows/main.yml`
-- [ ] Update DEVELOPERS.md with any project-specific requirements and commands
-- [ ] Update commands in `justfile`
+It is authenticated via a token signed by the configured `BACKEND_TOKEN`, which
+key shared between job-server and all services for a backend, and it uses the
```
Comment (ghickman) [inline README.md:10]: s/key/is/ ?

File: README.md [hunk 5]
```diff
@@ -1,20 +1,14 @@
...
+same token to authenticated against the job-server to upload files.
```
Comment (ghickman) [inline README.md:11]: s/authenticated/authenticate/

File: dev-token.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import sys
+from datetime import datetime, timedelta, timezone
+from urllib.parse import urljoin
+
+from hatch import config, signing
+
+
+def token(argv=sys.argv):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("workspace", help="workspace for token url")
```
Comment (ghickman) [inline dev-token.py:12]: The docs mention this isn't required but, and my argparse-memories are very rusty, isn't this a required argument?

File: dotenv-sample.env [hunk 7]
```diff
@@ -0,0 +1,7 @@
+BACKEND_TOKEN="really really really really really really really long secret"
+# local dev server
+SERVER_HOST=http://127.0.0.1:8001/
+# local job-server
+API_SERVER=http://127.0.0.1:8000/
+# workspaces dir
+WORKSPACES=./workspaces
```
Comment (ghickman) [inline dotenv-sample.env:7]: Could you pop a line before each comment? Helps to break up the text for anyone who struggles with lines merging 🙂

File: justfile [hunk 8]
```diff
@@ -50,6 +50,21 @@ update TARGET="prod":
...
 
+
 # Run the dev project
 run:
-    echo "Not implemented yet"
+    #!/usr/bin/env bash
+    set -euo pipefail
+    . scripts/setup_functions
+    dev_setup
+    port=$(echo $SERVER_HOST | awk -F: '{print $3}' | tr -d / )
```
Comment (ghickman) [inline justfile:60]: Does one not need to put quotes around the field separator?

File: justfile [hunk 9]
```diff
@@ -50,6 +50,21 @@ update TARGET="prod":
...
+    host=$(echo $SERVER_HOST | awk -F[:/] '{print $4}')
```
Comment (ghickman) [inline justfile:61]: Is this splititng on `:` or `/`? (ie regex multiple tokens syntax?)

**Review by ghickman — APPROVED** [review]
Great start!

Few small bits to tweak, could you also cut down the .gitignore to what we're actually using?  I missed that we'd used a giant template.  I'll go sort out the template repo's version soon too.

File: dev-token.py (hunk 6 above)
Comment (bloodearnest) [reply dev-token.py:12]: huh, yeah, I messed this up, will fix.

Not sure I want to allow an unrestricted token, so will required it

File: justfile (hunk 8 above)
Comment (bloodearnest) [reply justfile:60]: only if it needs quoting, eg `$`

File: justfile (hunk 9 above)
Comment (bloodearnest) [reply justfile:61]: yes, it's a regex hack to to split out the both the host part and the host:port part.

File: dev-token.py (hunk 6 above)
Comment (bloodearnest) [reply dev-token.py:12]: I removed the incorrect doc, workspace is required

**Comment by bloodearnest** [comment]
> Great start!
> 
> Few small bits to tweak, could you also cut down the .gitignore to what we're actually using? I missed that we'd used a giant template. I'll go sort out the template repo's version soon too.

Yeah, I kept the repos version for now. I'm not a fan of indescriminate ignoring, but I imagine some of those are useful on different platforms?

I'm tempted to leave it in, until we've merged an agreed default .gitignore into the template repo

**Review by ghickman — APPROVED** [review]

**Pull request merged**